Sources/RealModule/ApproximateEquality.swift

```diff
@@ -0,0 +1,111 @@
+  /// floating-point arithmetic. This comparison method may be helpful in
+  /// these situations.
+  ///
+  /// The simplest use of this method looks like:
+  /// ```
+  /// a.approximatelyEquals(b)
+  /// ```
+  /// when used like this, the result is true if `a ` and `b` agree to about
+  /// half the representable significant digits, which is the usual naive
+  /// guidance in numerical analysis ("if you don't have a careful analysis,
+  /// half your bits are probably bad").
+  ///
+  /// You can specify a tolerance to use for more specialized scenarios.
+  /// Three types of tolerance are supported:
+  ///
+  /// - `.absolute(t)`
+  ///   `a` is approximately equal to `b` with absolute tolerance `t` if
+  ///   ```
+  ///   (a - b).magnitude < t
+  ///   ```
+  ///   The trouble with an absolute tolerance is that it only makes sense
+  ///   when you know the expected scale of the `a` and `b`, but the *raison d'être*
+  ///   of floating-point numbers is to avoid scale dependent computations as much
+  ///   as possible. Because of this absolute tolerances are often a bad choice,
+  ///   unless you have very specific requirements.
+  ///
+  ///   There is no default absolute tolerance, because it's impossible for a general-purpose
+  ///   library to know what the scale of the values being compared might be.
+  ///
+  ///   However, when you do know the scale of an expected result, this can be an excellent
+  ///   choice. For example, suppose we want to repeat an iterative process until the result
+  ///   is within `0.1` of π:
+  ///   ```
+  ///   while !result.approximatelyEquals(.pi, tolerance: .absolute(0.1)) {
+  ///   }
+  ///   ```
+  ///   The tolerance `t` must be positive and finite.
+  ///
+  /// - `.relative(t)`
+  ///   `a` and `b` are approximately equal with relative tolerance `t` if
+  ///   ```
+  ///   let scale = max(a.magnitude, b.magnitude, .leastNormalMagnitude)
+  ///   return (a - b).magnitude < t * scale
+  ///   ```
+  ///   This type of tolerance matches well with the scale invariance of floating-point,
+  ///   making it a good choice for most problems. If you do not specify a tolerance,
+  ///   `approximatelyEquals` uses a relative tolerance of `sqrt(ulpOfOne)`.
+  ///
+  ///   The tolerance `t` must be in `.ulpOfOne ..< 1`.
+  ///
+  /// - `.relative(t, minimumScale: s)`
+  ///   In some cases it is desirable to override the `minimumScale` used in a relative
+  ///   comparison (either by setting it to zero, so that there is no rolloff at zero, or by setting it
+  ///   to a much larger value to account for cancellation). When a `minimumScale` is
+  ///   specified, the computation performed is:
+  ///   ```
+  ///   let scale = max(a.magnitude, b.magnitude, minimumScale)
+  ///   return (a - b).magnitude < t * scale
+  ///   ```
+  ///   The tolerance `t` must be in `.ulpOfOne ..< 1`.  The `minimumScale` must
+  ///   be positive and finite.
```

Comment:
Some folks may read "positive" as `> 0` — maybe use "greater than or equal to zero" or "non-negative" here?

Reply:
Positive actually means `> 0`, so they'd be correct. I can say "strictly positive" to make it clearer (or spell it as "greater than zero").